src/map/battle.c

```diff
@@ -1070,15 +1070,20 @@ static int64 battle_calc_cardfix(int attack_type, struct block_list *src, struct
 					}
 
 					// Apply bMagicSubDefEle damage reduction.
-					switch (src->type) {
-					case BL_MOB:
-						ele_fix += tsd->magic_sub_def_ele[status_get_element(src)].rate_mob;
-						break;
-					case BL_PC:
-						ele_fix += tsd->magic_sub_def_ele[status_get_element(src)].rate_pc;
-						break;
-					default:
-						break;
+					enum elements def_ele = status_get_element(src);
```

Comment:
if type is not mob or pc calling ``status_get_element`` is useless
probably move ``enum elements def_ele = status_get_element(src);`` under if?

Reply:
Fixed.
I'm using `sstatus->def_ele` instead of `status_get_element(src)` now.